src/systemctl/systemctl.c

```diff
@@ -399,7 +399,17 @@ static int output_units_list(const UnitInfo *unit_infos, unsigned c) {
         if (!table)
                 return log_oom();
 
+        r = table_set_display_all(table);
+        if (r < 0)
+                return log_error_errno(r, "Failed to set columns to display: %m");
+
         table_set_header(table, !arg_no_legend);
+        if (arg_no_legend) {
+                /* Hide the 'glyph' column when --no-legend is requested */
+                r = table_hide_column_from_display(table, (size_t) 0);
```

Comment:
I don't think the cast to (size_t) 0 is needed. In the `table_set_display` case it was needed because of varargs. Here, the compiler should do the upcast automatically based on the function signature.

Reply:
Makes sense.